staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered.go

```diff
@@ -157,9 +162,18 @@ func (b *bufferedBackend) Shutdown() {
 	// - When b.shutdownCh is closed, we know that the goroutine in Run has terminated.
 	// - This means that processIncomingEvents has terminated.
 	// - Which means that b.buffer is closed and cannot accept any new events anymore.
-	// - Because processEvents is called synchronously from the Run goroutine, the waitgroup has its final value.
-	// Hence wg.Wait will not miss any more outgoing batches.
-	b.wg.Wait()
+	// - Because processEvents is called synchronously from the Run goroutine, the b.ch has its final value.
+	// Hence wait for b.ch being empty will not miss any more outgoing batches.
+	if err := wait.PollImmediate(100*time.Microsecond, 60*time.Second, func() (bool, error) {
```

Comment:
Simple for to add `MaxGoRoutines` structs to the channel is enough, no need to re-implement the timeout with non-configurable parameters

Reply:
> Simple for to add MaxGoRoutines structs to the channel is enough

I donot  understand. How this logic can wait for all routines exit?

BTW, this poll reimplement the previous WG wait, only with a timeout.

Reply:
60s default to requestTimeout, it is long enough.